### DISCUSSION TITLE
'package:firebase_core/firebase_core.dart' => Target of URI doesn't exist ERROR

### DISCUSSION THREAD

**Discussion by EvoRye** [firebase/flutterfire#11432] [Q&A]
I have Firebase setup working, logged in, CLI is setup and working. The only thing that is giving me an error is: 'package:firebase_core/firebase_core.dart' Target of URI doesn't exist error (red squiggly line).

The dependency seems to be correct. I have tried every possible solution I could find on SO, Google, etc. Tried every chatGPT recommendation. Just cannot get the dependency to work.

Here is my pubspec.yaml:
![pubspec yaml screenshot](https://github.com/firebase/flutterfire/assets/117920279/79f7575a-264a-464b-806f-e2ca47e10bbe)

Here is my main.dart:

```
import 'package:flutter/material.dart';
import 'package:gogoapp/pages/login_page.dart';
import 'package:firebase_core/firebase_core.dart'; 
import 'firebase_options.dart';

void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await Firebase.initializeApp(
    options: DefaultFirebaseOptions.currentPlatform,
  );
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: LoginPage(),
    );
  }
}
```
Here are a few things I tried:
-flutter clean 
-flutter pub get 
-flutter channel (on stable) everything is latest version
-restart IDE and restart
-Windows Environment variables (Path) to Flutter and Dart setup. 

I've been stuck on this error for the past 3 days. Any suggestions?

Thanks

**Comment by RajamdGamer** [comment]
I don't know this would help but I just followed the instructions like:
1) Install Firebase CLI [here](https://firebase.google.com/docs/cli)
2) run 'dart pub global activate flutterfire_cli\ in your Flutter project
3) run 'flutterfire configure'
4) 'flutter pub add firebase_core'  

The following is my vs code internal Powershell terminal.
Learn more about using this file and next steps from the documentation:
 > https://firebase.google.com/docs/flutter/setup
PS G:\Flutter\Flutter Projects\FlutterXFirebase\notetakingapp> flutter packages get
Resolving dependencies... 
  flutter_lints 3.0.2 (4.0.0 available)
  leak_tracker 10.0.0 (10.0.5 available)
  leak_tracker_flutter_testing 2.0.1 (3.0.5 available)
  leak_tracker_testing 2.0.1 (3.0.1 available)
  material_color_utilities 0.8.0 (0.11.1 available)
  meta 1.11.0 (1.15.0 available)
  test_api 0.6.1 (0.7.1 available)
  vm_service 13.0.0 (14.2.2 available)
Got dependencies!
9 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
PS G:\Flutter\Flutter Projects\FlutterXFirebase\notetakingapp> flutter pub outdated
Showing outdated packages.
[*] indicates versions that are not the latest available.

Package Name   Current  Upgradable  Resolvable  Latest
direct dependencies: all up-to-date.

dev_dependencies:
flutter_lints  *3.0.2   *3.0.2      4.0.0       4.0.0

1 dependency is constrained to a version that is older than a resolvable version.
To update it, edit pubspec.yaml, or run `flutter pub upgrade --major-versions`.
PS G:\Flutter\Flutter Projects\FlutterXFirebase\notetakingapp> flutter packages upgrade
Resolving dependencies... (2.0s)
  flutter_lints 3.0.2 (4.0.0 available)
  leak_tracker 10.0.0 (10.0.5 available)
  leak_tracker_flutter_testing 2.0.1 (3.0.5 available)
  leak_tracker_testing 2.0.1 (3.0.1 available)
  material_color_utilities 0.8.0 (0.11.1 available)
  meta 1.11.0 (1.15.0 available)
  test_api 0.6.1 (0.7.1 available)
  vm_service 13.0.0 (14.2.2 available)
No dependencies changed.
9 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
PS G:\Flutter\Flutter Projects\FlutterXFirebase\notetakingapp> flutter pub outdated
Showing outdated packages.
[*] indicates versions that are not the latest available.

Package Name   Current  Upgradable  Resolvable  Latest  
direct dependencies: all up-to-date.

dev_dependencies:
flutter_lints  *3.0.2   *3.0.2      4.0.0       4.0.0

1 dependency is constrained to a version that is older than a resolvable version.
To update it, edit pubspec.yaml, or run `flutter pub upgrade --major-versions`.
PS G:\Flutter\Flutter Projects\FlutterXFirebase\notetakingapp> flutter pub upgrade --major-versions
Resolving dependencies... (1.8s)
> flutter_lints 4.0.0 (was 3.0.2)
  leak_tracker 10.0.0 (10.0.5 available)
  leak_tracker_flutter_testing 2.0.1 (3.0.5 available)
  leak_tracker_testing 2.0.1 (3.0.1 available)
> lints 4.0.0 (was 3.0.0)
  material_color_utilities 0.8.0 (0.11.1 available)
  meta 1.11.0 (1.15.0 available)
  test_api 0.6.1 (0.7.1 available)
  vm_service 13.0.0 (14.2.2 available)
Changed 2 dependencies!
7 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.

Changed 1 constraint in pubspec.yaml:
  flutter_lints: ^3.0.0 -> ^4.0.0
PS G:\Flutter\Flutter Projects\FlutterXFirebase\notetakingapp> 
 *  History restored 

Package flutterfire_cli is currently active at version 1.0.0.
The package flutterfire_cli is already activated at newest available version.
To recompile executables, first run `dart pub global deactivate flutterfire_cli`.
Installed executable flutterfire.
Activated flutterfire_cli 1.0.0.
PS G:\Flutter\Flutter Projects\FlutterXFirebase\notetakingapp> 
 *  History restored 

Computing fixes in notetakingapp (dry run)...

1 proposed fix in 1 file.

  unused_import • 1 fix

To fix an individual diagnostic, run one of:
  dart fix --apply --code=unused_import

To fix all diagnostics, run:
  dart fix --apply
Computing fixes in notetakingapp...
Applying fixes...

lib\main.dart
  unused_import • 1 fix

1 fix made in 1 file.
PS G:\Flutter\Flutter Projects\FlutterXFirebase\notetakingapp>  flutterfire configure
? You have an existing `firebase.json` file and possibly already configured your project for Firebase. Would you prefer to reuse the values in you? You have an existing `firebase.json` file and possibly already configured your project for Firebase. Would you prefer to reuse the values in you✔ You have an existing `firebase.json` file and possibly already configured your project for Firebase. Would you prefer to reuse the values in your existing `firebase.json` file to configure your project? · no 
? Generated FirebaseOptions file G:\Flutter\Flutter Projects\FlutterXFirebase\notetakingapp\lib\firebase_options.dart already exists, do you want 
? Generated FirebaseOptions file G:\Flutter\Flutter Projects\FlutterXFirebase\notetakingapp\lib\firebase_options.dart already exists, do you want 
✔ Generated FirebaseOptions file G:\Flutter\Flutter Projects\FlutterXFirebase\notetakingapp\lib\firebase_options.dart already exists, do you want 
to override it? · yes 
i Found 2 Firebase projects.
✔ Select a Firebase project to configure your Flutter application with · notesapp-1b1f4 (NotesApp) 
✔ Which platforms should your configuration support (use arrow keys & space to select)? · android, windows
i Firebase android app com.example.notetakingapp registered.
i Firebase windows app notetakingapp (windows) registered.


Learn more about using this file and next steps from the documentation:
 > https://firebase.google.com/docs/flutter/setup
PS G:\Flutter\Flutter Projects\FlutterXFirebase\notetakingapp> flutter pub add firebase_core 
Resolving dependencies... 
+ firebase_core 2.31.0
+ firebase_core_platform_interface 5.0.0
+ firebase_core_web 2.17.0
+ flutter_web_plugins 0.0.0 from sdk flutter
  leak_tracker 10.0.0 (10.0.5 available)
  leak_tracker_flutter_testing 2.0.1 (3.0.5 available)
  leak_tracker_testing 2.0.1 (3.0.1 available)
  material_color_utilities 0.8.0 (0.11.1 available)
  meta 1.11.0 (1.15.0 available)
+ plugin_platform_interface 2.1.8
  test_api 0.6.1 (0.7.1 available)
  vm_service 13.0.0 (14.2.2 available)
+ web 0.5.1
Changed 6 dependencies!
Try `flutter pub outdated` for more information.
Building with plugins requires symlink support.

Please enable Developer Mode in your system settings. Run
  start ms-settings:developers
to open settings.
PS G:\Flutter\Flutter Projects\FlutterXFirebase\notetakingapp> Run
>>   start ms-settings:developers
Run : The term 'Run' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or 
At line:1 char:1
+ Run
+ ~~~
    + CategoryInfo          : ObjectNotFound: (Run:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
 
PS G:\Flutter\Flutter Projects\FlutterXFirebase\notetakingapp> Run
>>   start ms-settings:developers
Run : The term 'Run' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or 
if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ Run
+ ~~~
    + CategoryInfo          : ObjectNotFound: (Run:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
 
PS G:\Flutter\Flutter Projects\FlutterXFirebase\notetakingapp> flutter pub add firebase_core 
"firebase_core" is already in "dependencies". Will try to update the constraint.
Resolving dependencies... 
  leak_tracker 10.0.0 (10.0.5 available)
  leak_tracker_flutter_testing 2.0.1 (3.0.5 available)
  material_color_utilities 0.8.0 (0.11.1 available)
  meta 1.11.0 (1.15.0 available)
  test_api 0.6.1 (0.7.1 available)
  vm_service 13.0.0 (14.2.2 available)
Got dependencies!
7 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
PS G:\Flutter\Flutter Projects\FlutterXFirebase\notetakingapp> flutter pub outdated
Showing outdated packages.
[*] indicates versions that are not the latest available.

Package Name  Current  Upgradable  Resolvable  Latest

direct dependencies: all up-to-date.

dev_dependencies: all up-to-date.
all dependencies are up-to-date.

**Reply by neonzhan** [reply]
This worked, thank you

**Comment by RajamdGamer** [comment]
if possible try running the commands in the terminal in order once , let me know if it got fixed
